src/GitHub.VisualStudio/Services/UsageService.cs

```diff
@@ -0,0 +1,136 @@
+{
+    [Export(typeof(IUsageService))]
+    public class UsageService : IUsageService
+    {
+        const string StoreFileName = "ghfvs.usage";
+        static readonly Calendar cal = CultureInfo.InvariantCulture.Calendar;
+        readonly IGitHubServiceProvider serviceProvider;
+        string storePath;
+
+        [ImportingConstructor]
+        public UsageService(IGitHubServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        public bool IsSameDay(DateTimeOffset lastUpdated)
+        {
+            return lastUpdated.Date == DateTimeOffset.Now.Date;
+        }
+
+        public bool IsSameWeek(DateTimeOffset lastUpdated)
+        {
+            return GetIso8601WeekOfYear(lastUpdated) == GetIso8601WeekOfYear(DateTimeOffset.Now);
+        }
+
+        public bool IsSameMonth(DateTimeOffset lastUpdated)
+        {
+            return lastUpdated.Month == DateTimeOffset.Now.Month;
+        }
+
+        public IDisposable StartTimer(Func<Task> callback, TimeSpan dueTime, TimeSpan period)
+        {
+            return new Timer(
+                async _ =>
+                {
+                    try { await callback(); }
+                    catch { /* log.Warn("Failed submitting usage data", ex); */ }
+                },
+                null,
+                dueTime,
+                period);
+        }
+
+        public async Task<UsageData> ReadLocalData()
+        {
+            await Initialize();
+
+            var json = File.Exists(storePath) ? await ReadAllTextAsync(storePath) : null;
+
+            try
+            {
+                return json != null ?
+                    SimpleJson.DeserializeObject<UsageData>(json) :
+                    new UsageData { Model = new UsageModel() };
+            }
+            catch
+            {
+                return new UsageData { Model = new UsageModel() };
+            }
+        }
+
+        public async Task WriteLocalData(UsageData data)
+        {
+            try
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(storePath));
+                var json = SimpleJson.SerializeObject(data);
+                await WriteAllTextAsync(storePath, json);
+            }
+            catch
+            {
+                // log.Warn("Failed to write usage data", ex);
+            }
+        }
+
+        async Task Initialize()
+        {
+            if (storePath == null)
+            {
+                await ThreadingHelper.SwitchToMainThreadAsync();
+
+                var program = serviceProvider.GetService<IProgram>();
+                storePath = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                    program.ApplicationName,
+                    StoreFileName);
+            }
+        }
+
+        async Task<string> ReadAllTextAsync(string path)
+        {
+            using (var s = File.OpenRead(path))
```

Comment:
You can use rothko for IO services that can be mocked for unit testing

Reply:
Already thought of that, but Rothko doesn't have the methods we need as far as I can tell, which is partly why I split this I/O functionality out into a separate service apart from the usage tracker itself.